provisioning/wildfly_channel_in_zips.adoc

```diff
@@ -0,0 +1,210 @@
+
+=== Implementation Plan
+
+The channel metadata are stored in the `.installation` directory when a WildFly server is provisioned.
+The WildFly distributions (zip & tar.fz archives) will contain this directory.
+This is covered by https://issues.redhat.com/browse/WFLY-19221[WFLY-19221] and provides the foundation to update their installations.
+
+Prospero will be updated to cover the additional use cases from this proposal.
+In particular, it would have to be able to let a user chose a new "target" channel to update to.
+This is covered by https://github.com/wildfly-extras/prospero/issues/749.
+
+==== Galleon Universe
+
+WildFly provisioning metadata contains a recording of the Galleon provisioning state in  `.installation/provisioning_record.xml`.
+
+Feature Packs are identified with Galleon feature pack location (e.g. `wildfly@maven(org.jboss.universe:community-universe):current#33.0.0.Final`) as described in the https://docs.wildfly.org/galleon/#_feature_pack_location[Galleon documentation].
+
+Prospero is not able to "understand" this location without access to the universe artifacts (`org.jboss.universe:community-universe` & `org.jboss.universe.producer:wildfly-producers`).
+The WildFly producer is frequently updated to add new major versions of WildFly including the upcoming major version.
+By adding these artifacts to the WildFly channel manifest, Prospero will be able to resolve the feature pack location and ultimately point to the appropriate Maven artifact for the feature pack.
+
+== Backwards Compatibility
+
+This enhancement does not affect previous versions of WildFly.
+
+=== Default Configuration
+
+Updating an installation could update its default configuration (e.g. if the update is to a major version).
+
+WildFly does not make guarantee that its default configuration will be compatible from one major version to another (even though in practice, that's often the case).
+
+
+=== Importing Existing Configuration
+
+This section is not relevant in the context of this enhancement.
+
+=== Deployments
+
+Deployments can be affected by this enhancement if there are incompatibilities between the existing WildFly installation and the update.
+
+Testing an installation is out of scope of this enhancement but should be incorporated in the user workflow (e.g. to test the update in a staging environment).
+
+=== Interoperability
+
+This section is not relevant in the context of this enhancement.
+
+== Test Plan
+
+=== Test Plan for WFLY-19221 - [Preview] Incorporate channel metadata in the download zips
+
+* Verify that WildFly generated distributions (from the `dist`, `ee-dist`, and `preview-dist` Maven Modules) contain the channel metadata files corresponding to their provisioning states.
+
```

Comment:
We have the testsuite/galleon tests which test updating an installation using Galleon tooling. It seems like we could expand this to cover this update mechanism.